core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -160,46 +173,60 @@ func (c *OutgoingConnectorHandler) AwaitConnection(ctx context.Context) (string,
 				return "", fmt.Errorf("failed to select gateway: %w", err)
 			}
 
-			// Cycling through gateways via round robin, if we have attempted this connection before, then we have
-			// seen them all, backoff before starting over.
+			md.gateway = gateway
+
 			if attempts[gateway] > 0 {
-				c.lggr.Warnw("all available gateway nodes attempted without connection, backing off", "waitTime", wait)
-				attempts = make(map[string]int)
-
-				// hold until wait or context expires
-				select {
-				case <-ctx.Done():
-					return "", ctx.Err()
-				case <-time.After(wait):
```

Comment:
is the intention here to retry just 1 gateway after backoff or did you want to retry all?  Without resetting attempts after the first backoff it will then only try 1 before backing off again.